org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -533,150 +533,148 @@ private boolean checkPaused() throws Exception {
-			throw sne;
-		} catch (Exception ne) {
-			throw new ScanningException(ne);
-		}
+		work(()-> pauseInternal(), DeviceState.RUNNING, "pause", true, false);
+	}
+	private void pauseInternal() throws ScanningException, InterruptedException {
 		
 		getBean().setPreviousStatus(getBean().getStatus());
 		getBean().setStatus(Status.PAUSED);
 		setDeviceState(DeviceState.SEEKING);
-		try {
-			awaitPaused = true;
-			if (getModel().getDetectors()!=null) for (IRunnableDevice<?> device : getModel().getDetectors()) {
-				DeviceState currentState = device.getDeviceState();
-				if (currentState.isRunning()) {
-					if (device instanceof IPausableDevice) {
-						((IPausableDevice)device).pause();
-					}
-				} else {
-					logger.info("Device " + device.getName() + " wasn't running to pause. Was + " + currentState);
+		if (getModel().getDetectors()!=null) for (IRunnableDevice<?> device : getModel().getDetectors()) {
+			DeviceState currentState = device.getDeviceState();
+			if (currentState.isRunning()) {
+				if (device instanceof IPausableDevice) {
+					((IPausableDevice<?>)device).pause();
 				}
+			} else {
+				logger.info("Device " + device.getName() + " wasn't running to pause. Was + " + currentState);
 			}
-			setDeviceState(DeviceState.PAUSED);
-			
-		} catch (ScanningException s) {
-			throw s;
-		} catch (Exception ne) {
-			throw new ScanningException(ne);
-		} finally {
-			lock.unlock();
 		}
+		setDeviceState(DeviceState.PAUSED);
 	}
 	
 	@Override
-	public void seek(int stepNumber) throws ScanningException, InterruptedException {
+	public void seek(final int stepNumber) throws ScanningException, InterruptedException {
 		// This is the values of all motors at this global (including malcolm) scan
 		// position. Therefore we do not need a subscan moderator but can run the iterator
 		// to the point
+		work(()-> seekInternal(stepNumber), DeviceState.PAUSED, "seek", true, false);
+	}
+	private void seekInternal(int stepNumber) throws ScanningException, InterruptedException {
+		
 		if (stepNumber<0) throw new ScanningException("Seek position is invalid "+stepNumber);
 		if (stepNumber>location.getTotalSize())  throw new ScanningException("Seek position is invalid "+stepNumber);
-
-		// We just changed the location from where we run the scan from and require a new location.
 		this.positionIterator = location.createPositionIterator();
 		IPosition pos = location.seek(stepNumber, positionIterator);
 		positioner.setPosition(pos);
 		if (getModel().getDetectors()!=null) for (IRunnableDevice<?> device : getModel().getDetectors()) {
-			if (device instanceof IPausableDevice) ((IPausableDevice)device).seek(stepNumber);
+			if (device instanceof IPausableDevice) ((IPausableDevice<?>)device).seek(stepNumber);
 		}
 	}
 
 	@Override
-	public void resume() throws ScanningException {
-		
+	public void resume() throws ScanningException, InterruptedException {
+		work(()-> resumeInternal(), DeviceState.PAUSED, "resume", false, true);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)